### DISCUSSION TITLE
Push button

### DISCUSSION THREAD

**Discussion by jnaor** [foxglove/community#13] [Q&A] [answered]
I'm using rviz as a simple GUI with a few actions triggered by the user pressing a push button (an rviz custom panel which publishes a ROS2 topic). I'm thinking of migrating to FoxGlove and didn't understand from the documentation if this functionality is available. Would appreciate any information.
Thanks!

**Comment by jtbandes** [comment, marked as answer]
Try out the Publish panel: https://foxglove.dev/docs/studio/panels/publish

**Comment by defunctzombie** [comment]
You can also write a custom extension that can use the `publish` api (assuming you are connected to a data source that supports publishing).

**Comment by jnaor** [comment]
Thanks @jtbandes and @defunctzombie for your answers. To clarify my question, I need the PushButton UI functionality (the graphic element itself), which then triggers a ROS2 message. A custom extension sounds like the way to go (in rviz I used a "custom panel" which uses Qt for the graphic elements), but I haven't found this functionality in the documentation.

**Reply by defunctzombie** [reply]
The docs for extensions are here: https://foxglove.dev/docs/studio/extensions/getting-started

**Reply by tmp-username** [reply]
It looks like there is no interface like Rviz's InteractiveMarkerServer, so have you found a good solution? @jnaor

**Comment by jnaor** [comment]
I haven't...
